README.md

```diff
@@ -141,15 +142,130 @@ for (index, game) in myFavoriteZeldaGames.enumerate() {
+
+This last example introduced a couple of new concepts, if you want to understand better what a line of code is doing then feel free to add `print` and `if` statements to inspect values (this is called 'debugging'). For example to see the value of `squaredNumber` you could add `print (squaredNumber)` inside the loop.
+
+Maybe one of your friends deserves a special mention for being so awesome, so you need to find them in the `friends` list, and when you do, print a message for them:
+
+```swift
+let friends = ["Chris", "Billy", "Jessica"]
+
+for friend in friends {
+    print("We are inside the for loop, friend: \(friend)")
+
+    if friend == "Billy" {
+        print("Hello \(friend), you are awesome!")
+    }
+}
+
+// We are inside the for loop, friend: Chris
+// We are inside the for loop, friend: Billy
+// Hello Billy, you are awesome!
+// We are inside the for loop, friend: Jessica
+```
+
+In the above, you iterate the `friends` list printing the value of each index in the array. If the current value is equal to "Billy", then inside an if statement you print them a special greeting.
+
+## Loops and Functions
+
+In unit 1, lesson 8 you learned about functions, a convenient way to package functionality into reusable components. You can add any code can into a function, including loops.
+
+Say for example you want to know how many of your friends got over 70 in their recent exam.
+
+You first create an array of their grades and pass that array to a function. That function iterates through the array and maintains a count of how many grades were above 70, and returns that value.
+
+```swift
+func numberOfFriendsAbove70(grades: [Int]) -> Int {
+    var studentsAbove70 = 0
+
+    for grade in grades {
+        if grade >= 70 {
+            studentsAbove70 += 1
+        }
+    }
+
+    return studentsAbove70
+}
+
+let gradesFromExam = [80, 45, 79, 32, 84, 100, 53, 60]
+
+print numberOfFriendsAbove70(gradesFromExam)
+// 4
 ```
 
-* If we do cover enumerate, maybe we VERY briefly talk about Tuples (but not too much because Tuples will be brought up in another unit.
- 
+Again, there's a lot happening in this example, so let's break everything down by starting at the bottom of the code. First you create an array called `gradesFromExam` of your friends grades that can't change and then pass that array to the `numberOfFriendsAbove70` function and print the returned result.
+
+At the declaration of the `numberOfFriendsAbove70` function you set the incoming parameter as an array of numbers and the return value as a single number. Inside the function you first initialize to 0 a variable to count the number of grades over 70.
 
+Next iterate through each value in the `grades` array and if the value at the current index is greater than 70 increase the value of `studentsAbove70` by 1\. Once the array has been fully iterated, return the total value or `studentsAbove70`.
 
```

Comment:
I would add one more note at the very end here challenging the student to run through this code in their head to see if they can correctly step through each line of code with an understanding of what's happening.

Then I would challenge the student to re-create this problem solved here in a playground file, adding print functions to see when this stuff gets printed to see if how they thought it would play out was indeed correct.